src/components/views/right_panel/RoomHeaderButtons.tsx

```diff
@@ -158,18 +159,18 @@ export default class RoomHeaderButtons extends HeaderButtons<IProps> {
 
     public renderButtons() {
         return <>
-            <PinnedMessagesHeaderButton
+            { !this.props.coreElementsOnly && <PinnedMessagesHeaderButton
                 room={this.props.room}
                 isHighlighted={this.isPhase(RightPanelPhases.PinnedMessages)}
                 onClick={this.onPinnedMessagesClicked}
-            />
-            <TimelineCardHeaderButton
+            /> }
+            { this.props.coreElementsOnly && <TimelineCardHeaderButton
```

Comment:
It's a bit awkward to use `coreElementsOnly` as the signal to **both** filter out non-core and **also add** the timeline for maximised widgets. Maybe this part here takes a separate prop about whether there's a maximised widget displayed? That would make the logic more obvious.

Reply:
Just to clarify:
Two props: `coreElementOnly` + `showTimelineCardHeaderButton`
Or one porp like now but renamed to sth like: `nonTimelineMainSplit`

Reply:
Seems like you have found a good way with the excluded set. 😄